lib/widgets/recent_dm_conversations.dart

```diff
@@ -0,0 +1,119 @@
+
+  @override
+  void onNewStore() {
+    model?.removeListener(_modelChanged);
+    model = PerAccountStoreWidget.of(context).recentDmConversationsView
+      ..addListener(_modelChanged);
+  }
+
+  void _modelChanged() {
+    setState(() {
+      // The actual state lives in [model].
+      // This method was called because that just changed.
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final sorted = model!.sorted;
+    return Scaffold(
+      appBar: AppBar(title: const Text('Direct messages')),
+      body: ListView.builder(
+        itemCount: sorted.length,
+        itemBuilder: (context, index) => RecentDmConversationsItem(narrow: sorted[index])));
+  }
+}
+
+class RecentDmConversationsItem extends StatelessWidget {
+  const RecentDmConversationsItem({super.key, required this.narrow});
+
+  final DmNarrow narrow;
+
+  final _avatarSize = 32.0;
+  final _avatarBorderRadius = 3.0;
+
+  @override
+  Widget build(BuildContext context) {
+    final store = PerAccountStoreWidget.of(context);
+    final selfUser = store.users[store.account.userId]!;
+
+    final String title;
+    final Widget avatar;
+    switch (narrow.otherRecipientIds) {
+      case []:
+        title = selfUser.fullName;
+        avatar = Avatar(userId: selfUser.userId, size: _avatarSize, borderRadius: _avatarBorderRadius);
+      case [var otherUserId]:
+        final otherUser = store.users[otherUserId];
+        title = otherUser?.fullName ?? '(unknown user)';
+        avatar = otherUser != null
+          ? Avatar(userId: otherUser.userId, size: _avatarSize, borderRadius: _avatarBorderRadius)
+          : SizedBox.square(dimension: _avatarSize);
+      default:
+        // TODO(i18n): List formatting, like you can do in JavaScript:
+        //   new Intl.ListFormat('ja').format(['Chris', 'Greg', 'Alya'])
+        //   // 'Chris、Greg、Alya'
+        title = narrow.otherRecipientIds.map((id) => store.users[id]!.fullName).join(', ');
+        avatar = SizedBox.square(
+          dimension: _avatarSize,
+          child: ClipRRect(
+            borderRadius: BorderRadius.all(Radius.circular(_avatarBorderRadius)),
+            clipBehavior: Clip.antiAlias,
+            child: ColoredBox(color: const Color(0x33808080),
+              child: Center(
+                child: Icon(ZulipIcons.group_dm, color: Colors.black.withOpacity(0.5))))));
+    }
+
+    return InkWell(
+      onTap: () {
+        Navigator.push(context,
+          MessageListPage.buildRoute(context: context, narrow: narrow));
+      },
+      child: ConstrainedBox(constraints: const BoxConstraints(minHeight: 48),
+        child: Row(crossAxisAlignment: CrossAxisAlignment.center, children: [
+          Padding(padding: const EdgeInsets.fromLTRB(12, 8, 0, 8),
+            child: avatar),
+          const SizedBox(width: 8),
+          Expanded(child: Padding(
+            padding: const EdgeInsets.symmetric(vertical: 4),
+            child: Text(
+              title,
```

Comment:
nit: I think the natural analogue of putting a widget's `child` last is to put a `Text` widget's positional String argument last.

This only became possible with a relatively recent version of Dart, circa 2022, so there may be a lot of code even in the Flutter tree that doesn't. But I think it's the natural thing to do now that one can.